### PR TITLE
Update `fltk` library to version 1.3.8

### PR DESCRIPTION
**Pull request by pinkahd**
Fix build on macOS 12.3.1, by updating the `fltk` library
to version 1.3.8. After the update I was able to compile
the game and play it again.

After looking at the changelog the majority of changes
are fixes for macOS versions.

https://github.com/fltk/fltk/blob/release-1.3.8/CHANGES

---

This issue occurred to me after I've updated to a newer
macOS version.

### CODE REVIEW

File: dependencies/lib-fltk/builder/CMakeLists.txt.in [hunk 1]
```diff
@@ -33,7 +33,7 @@ externalproject_add(fltk
         -DOPTION_USE_XFT=OFF
         -DOPTION_USE_XRENDER=OFF
         -DOPTION_USE_XDBE=OFF
-        -DOPTION_BUILD_EXAMPLES=OFF
```
Comment (lynxlynxlynx) [inline dependencies/lib-fltk/builder/CMakeLists.txt.in:36]: This flag got removed upstream?

**Comment by lynxlynxlynx** [comment]
The mingw build failed to link properly:
```make
 lib-fltk/lib/libfltk.a(Fl.cxx.obj): In function `WndProc':
/home/runner/work/ja2-stracciatella/ja2-stracciatella/ci-build/dependencies/lib-fltk/builder/fltk-prefix/src/fltk/src/Fl_win32.cxx:1217: undefined reference to `__imp__TrackMouseEvent'
collect2: error: ld returned 1 exit status
CMakeFiles/ja2-launcher.dir/build.make:209: recipe for target 'ja2-launcher.exe' failed
```

File: dependencies/lib-fltk/builder/CMakeLists.txt.in (hunk 1 above)
Comment (pinkahd) [reply dependencies/lib-fltk/builder/CMakeLists.txt.in:36]: It got obsolete, I'm getting this when reverting to the previous version.

```
CMake Warning at CMake/options.cmake:113 (message):
  'OPTION_BUILD_EXAMPLES' is obsolete, please use 'FLTK_BUILD_TEST' instead.
Call Stack (most recent call first):
  CMakeLists.txt:79 (include)
```

**Comment by pinkahd** [comment]
> ```makefile
> error: ld
> ```

I'm not sure what happened here, the only related issue I could find was https://github.com/fltk/fltk/pull/171 but that seams to be related to version 1.1.x of the library.

Thinking of one solution here to use the old library for mingw64 platform as a fix. Open to other suggestions for a possible fix here since I'm not that familiar with `mingw64`.

**Comment by lynxlynxlynx** [comment]
Looks like a missing include in the launcher to me.

**Comment by selaux** [comment]
The _TrackMouseEvent function is part in the comctl32.dll library so we seem to need to link with comctl32.lib now. Its probably sufficient to add `comctl32` [here](https://github.com/ja2-stracciatella/ja2-stracciatella/blob/master/CMakeLists.txt#L279).

**Comment by pinkahd** [comment]
> The _TrackMouseEvent function is part in the comctl32.dll library so we seem to need to link with comctl32.lib now. Its probably sufficient to add `comctl32` [here](https://github.com/ja2-stracciatella/ja2-stracciatella/blob/master/CMakeLists.txt#L279).

Thanks, I've added `comctl32` as a dependency in [4cba23a](https://github.com/ja2-stracciatella/ja2-stracciatella/pull/1500/commits/4cba23ab62590613b60089d9fb29def8d1442866) 🤞🏼

**Comment by lynxlynxlynx** [comment]
The error is still the same, since you added the link flag to ja2.exe, not the launcher a few lines below.

**Comment by pinkahd** [comment]
> The error is still the same, since you added the link flag to ja2.exe, not the launcher a few lines below.

My mistake appended those to `LAUNCHER_LIBRARIES` now.

**Comment by lynxlynxlynx** [comment]
Hmm, maybe the ordering matters and it needs to be in the link line before fltk itself? 

Or we should just fix the fltk build instead. Its pkgconfig, cmake target info or whatever it exports should also include comctl32. Not sure if it can be included directly in the .a(r) file.

**Comment by pinkahd** [comment]
> Hmm, maybe the ordering matters and it needs to be in the link line before fltk itself?
> 
> Or we should just fix the fltk build instead. Its pkgconfig, cmake target info or whatever it exports should also include comctl32. Not sure if it can be included directly in the .a(r) file.

This is a bit out of my area of expertise (I rarely touch cmake). 

But I'm happy to give it a try, it would be useful if there's a way to run `mingw64` build on macOS or if I can run this within docker. Just want something to reproduce the issue locally instead of relying on the CI.

**Comment by lynxlynxlynx** [comment]
I meant something like `target_link_libraries(${LAUNCHER_BINARY} -lcomctl32 ${FLTK_LIBRARIES} ${STRACCIATELLA_LIBRARIES} ${LAUNCHER_LIBRARIES})`.

**Comment by pinkahd** [comment]
> I meant something like `target_link_libraries(${LAUNCHER_BINARY} -lcomctl32 ${FLTK_LIBRARIES} ${STRACCIATELLA_LIBRARIES} ${LAUNCHER_LIBRARIES})`.

I've added an if/else for `mingw64` and added `lcomctl32` earlier. Let's see 🤞🏼

**Comment by lynxlynxlynx** [comment]
Next thing that comes to mind is trying the same, but in fltk's cmake file at `dependencies/lib-fltk/CMakeLists.txt`.

If that won't be enough, I guess we'll have to use PATCH_COMMAND of ExternalProject to make changes to the source. Or perhaps it can be done in the parent previous CMakeLists.txt in between the two invocations.
https://cmake.org/cmake/help/latest/module/ExternalProject.html

**Comment by selaux** [comment]
I could reproduce this locally and the following worked for me as described [here](https://github.com/fltk/fltk/blob/f5628aa66de0173f4fc145586a8c83e66e32f160/src/Fl_win32.cxx#L215):

In `dependencies/lib-fltk/builder/CMakeLists.txt.in` replace the line

```
        "-DCMAKE_CXX_FLAGS=@CMAKE_CXX_FLAGS@"
```

with

```
        "-DCMAKE_CXX_FLAGS=@CMAKE_CXX_FLAGS@ -DNO_TRACK_MOUSE"
```

This will avoid linking the function altogether. I also tried various ways to link the library, but basically nothing worked.

**Comment by pinkahd** [comment]
> I could reproduce this locally and the following worked for me as described [here](https://github.com/fltk/fltk/blob/f5628aa66de0173f4fc145586a8c83e66e32f160/src/Fl_win32.cxx#L215):
> 
> In `dependencies/lib-fltk/builder/CMakeLists.txt.in` replace the line
> 
> ```
>         "-DCMAKE_CXX_FLAGS=@CMAKE_CXX_FLAGS@"
> ```
> 
> with
> 
> ```
>         "-DCMAKE_CXX_FLAGS=@CMAKE_CXX_FLAGS@ -DNO_TRACK_MOUSE"
> ```
> 
> This will avoid linking the function altogether. I also tried various ways to link the library, but basically nothing worked.

Thanks Stefan 🙌, pushed another commit with the fix 🤞🏻

**Review by lynxlynxlynx — APPROVED** [review]

**Comment by selaux** [comment]
@pinkahd Thanks for your effort!

**Pull request merged**